# DATA100 - Final Project: Storytelling with Data 

<hr>
<hr>

## Group 3 Members 
* Argonza, Antoinette Joy 
* Jamia, Gillian Nicole 
* Magsano, Niño Matthew 
* Reyes, Anton Gabriel

<hr>
<hr>

## Motivation

*As Lasallian students, we would want to determine the possible causes of child mortality and provide credible, consolidated information and insights that help prevent or solve this pressing social issue.*


<hr>
<hr>

## Libraries, Packages, or Modules

In [467]:
import os 
import csv
import time
import numpy as np
import pandas as pd
import seaborn as sns
import requests 
import datetime as dt
import geopandas as gpd
import matplotlib.pyplot as plt

from shapely.geometry import Point
from IPython.core.display import HTML

%matplotlib inline
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [468]:
plt.style.use('seaborn-whitegrid')

<hr>
<hr>

## Data Collection

### Datasets

**Initial datasets:** <br>
Upon initial data collection and reading of datasets, these databases were dropped, but it is still listed here to show our initial trajectory towards the storytelling. The data collection codes for these are removed in the notebook to make it presentable. <br>
* (Removed) The dataset for **Mortality among children** can be found here: [`Kaggle` Dataset Source](https://www.kaggle.com/mpwolke/cusersmarildownloadsdeathscsv) <br>
* (Removed) The dataset for **WHO - Immunization coverage estimates by country** can be found here: [`Kaggle` Dataset Source](https://www.kaggle.com/lsind18/who-immunization-coverage) <br>
* (Removed) The dataset for **Child Health Dataset** can be found here: [`Kaggle` Dataset Source](https://www.kaggle.com/hijest/child-health-dataset-who) <br>
* (Removed) The dataset for **Out of School Rates Global Data** can be found here: [`Kaggle` Dataset Source](https://www.kaggle.com/komalkhetlani/out-of-school-rates-global-data?select=Primary.csv) <br>
* (Removed) The dataset for **World Health Statistics 2020** can be found here: [`Kaggle` Dataset Source](https://www.kaggle.com/utkarshxy/who-worldhealth-statistics-2020-complete?select=adolescentBirthRate.csv) <br>

**Official datasets**: <br>
* The dataset for **Mortality rate, under-5 (per 1,000 live births)** can be found here: [`The World Bank` Dataset Source](https://data.worldbank.org/indicator/SH.DYN.MORT?end=2019&fbclid=IwAR1SIiyIcig6Mwin1-szdOQoKMCC6BMJrb0NrdbS1-bnL8gd2JoalibPjYI&start=1960) <br>
* The dataset for **Causes of death in children under 5** can be found here: [`Our World in Data` Dataset Source](https://ourworldindata.org/grapher/causes-of-death-in-children-under-5) <br>
* The dataset for **Malnutrition across the globe** can be found here: [`Kaggle` Dataset Source](https://www.kaggle.com/ruchi798/malnutrition-across-the-globe) <br>

**Helpful dataset/s**: <br>
These/This are/is included for possible merges of cleaned databases for data visualization and analysis. <br>
* The dataset for **Country Codes Alpha-2 & Alpha-3** can be found here: [`IBAN` Dataset Source](https://www.iban.com/country-codes) <br>

<hr>
<hr>

## Data Wrangling & Exploratory Data Analysis

<hr>

### Mortality rate, under-5 (per 1,000 live births)

In [469]:
mortality = pd.read_csv('Databases/mortality_rate.csv')
mortality.head()

Country Name Country Code  \
0                        Aruba          ABW   
1  Africa Eastern and Southern          AFE   
2                  Afghanistan          AFG   
3   Africa Western and Central          AFW   
4                       Angola          AGO   

                                    Indicator Name Indicator Code  1960  1961  \
0  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT   NaN   NaN   
1  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT   NaN   NaN   
2  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT   NaN   NaN   
3  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT   NaN   NaN   
4  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT   NaN   NaN   

    1962   1963        1964        1965  ...        2011        2012  \
0    NaN    NaN         NaN         NaN  ...         NaN         NaN   
1    NaN    NaN         NaN         NaN  ...   82.105187   78.354228   
2  344.6  338.7  333.100000  327.600000  ...   83.900000   80.300000   
3    NaN    NaN  308.353805  302.620962  ...  118.442691  115.498791   
4    NaN    NaN         NaN         NaN  ...  112.300000  105.000000   

         2013        2014        2015        2016        2017       2018  \
0         NaN         NaN         NaN         NaN         NaN        NaN   
1   74.991597   71.996841   69.288947   66.667614   64.347025  62.115387   
2   76.800000   73.600000   70.400000   67.600000   64.900000  62.500000   
3  112.667923  110.264426  107.765705  105.055552  102.430680  99.598781   
4   98.600000   93.000000   88.200000   84.200000   80.600000  77.700000   

        2019  2020  
0        NaN   NaN  
1  60.098659   NaN  
2  60.300000   NaN  
3  96.814240   NaN  
4  74.700000   NaN  

[5 rows x 65 columns]

In [470]:
mortality.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266 entries, 0 to 265
Data columns (total 65 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    266 non-null    object 
 1   Country Code    266 non-null    object 
 2   Indicator Name  266 non-null    object 
 3   Indicator Code  266 non-null    object 
 4   1960            112 non-null    float64
 5   1961            112 non-null    float64
 6   1962            116 non-null    float64
 7   1963            119 non-null    float64
 8   1964            124 non-null    float64
 9   1965            127 non-null    float64
 10  1966            129 non-null    float64
 11  1967            133 non-null    float64
 12  1968            137 non-null    float64
 13  1969            142 non-null    float64
 14  1970            144 non-null    float64
 15  1971            149 non-null    float64
 16  1972            153 non-null    float64
 17  1973            153 non-null    flo

In [471]:
mortality.describe()

1960        1961        1962        1963        1964        1965  \
count  112.000000  112.000000  116.000000  119.000000  124.000000  127.000000   
mean   151.166071  146.613393  149.111207  146.794958  146.716563  145.343472   
std     95.556097   94.282688   98.883656   99.767226   99.983448  100.351548   
min     19.600000   19.200000   18.600000   17.900000   17.100000   16.300000   
25%     66.975000   64.450000   68.550000   64.300000   63.700000   62.500000   
50%    149.400000  146.500000  142.950000  137.100000  133.150000  127.900000   
75%    220.925000  215.900000  214.525000  210.900000  218.350000  215.700000   
max    391.700000  386.300000  409.800000  420.600000  413.500000  406.500000   

             1966        1967        1968        1969  ...        2011  \
count  129.000000  133.000000  137.000000  142.000000  ...  241.000000   
mean   143.094516  139.498974  134.915993  132.530716  ...   38.006505   
std     98.926580   97.909379   96.274734   95.129489  ...   36.033354   
min     15.500000   14.900000   14.400000   13.900000  ...    2.500000   
25%     62.000000   60.500000   56.700000   55.775000  ...   10.000000   
50%    125.000000  119.100000  114.800000  108.550000  ...   22.400000   
75%    212.600000  210.600000  200.700000  201.150000  ...   59.155108   
max    399.900000  393.400000  387.300000  381.400000  ...  153.200000   

             2012        2013        2014        2015        2016        2017  \
count  241.000000  241.000000  241.000000  241.000000  241.000000  241.000000   
mean    36.644066   35.375455   34.254044   33.178938   32.107112   31.126408   
std     34.745564   33.555812   32.563909   31.620253   30.478658   29.595854   
min      2.400000    2.200000    2.100000    2.000000    1.900000    1.800000   
25%      9.800000    9.300000    8.900000    8.300000    8.000000    7.700000   
50%     21.800000   21.100000   20.600000   19.700000   19.100000   18.600000   
75%     56.800000   54.600000   52.600000   50.800000   49.000000   47.300000   
max    146.600000  142.100000  137.700000  138.300000  128.400000  124.400000   

             2018        2019  2020  
count  241.000000  241.000000   0.0  
mean    30.145293   29.214801   NaN  
std     28.698271   27.841743   NaN  
min      1.700000    1.700000   NaN  
25%      7.400000    7.000000   NaN  
50%     18.100000   17.300000   NaN  
75%     45.600000   44.200000   NaN  
max    120.300000  117.200000   NaN  

[8 rows x 61 columns]

In [472]:
mortality.describe(include=object)

Country Name Country Code  \
count           266          266   
unique          266          266   
top           Aruba          ABW   
freq              1            1   

                                         Indicator Name Indicator Code  
count                                               266            266  
unique                                                1              1  
top     Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  
freq                                                266            266

In [473]:
mortality.isnull().sum()*100/mortality.shape[0]

Country Name        0.000000
Country Code        0.000000
Indicator Name      0.000000
Indicator Code      0.000000
1960               57.894737
                     ...    
2016                9.398496
2017                9.398496
2018                9.398496
2019                9.398496
2020              100.000000
Length: 65, dtype: float64

In [474]:
mortality.duplicated().sum()

0

In [475]:
mortality.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004',
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014', '2015', '2016', '2017', '2018', '2019', '2020'],
      dtype='object')

#### Since 2020 has no observations and the years below 2000 will not provide any sufficient recent conclusions and recommendations other than provide a downward trend, these columns or variables will be dropped to accomodate other databases which has lesser years as well. 

In [476]:
mortality.drop(['1960', '1961', '1962', '1963', '1964', '1965', '1966', '1967', '1968',
       '1969', '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977',
       '1978', '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986',
       '1987', '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995',
       '1996', '1997', '1998', '1999', '2020'], axis=1, inplace=True)

mortality.head()

Country Name Country Code  \
0                        Aruba          ABW   
1  Africa Eastern and Southern          AFE   
2                  Afghanistan          AFG   
3   Africa Western and Central          AFW   
4                       Angola          AGO   

                                    Indicator Name Indicator Code        2000  \
0  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT         NaN   
1  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  136.917815   
2  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  128.700000   
3  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  169.548432   
4  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  203.900000   

         2001        2002        2003        2004        2005  ...  \
0         NaN         NaN         NaN         NaN         NaN  ...   
1  131.670422  126.139085  120.635993  115.263632  110.060055  ...   
2  124.600000  120.400000  116.300000  112.100000  107.900000  ...   
3  164.533824  159.251950  153.916285  148.577352  143.417384  ...   
4  197.800000  190.900000  183.200000  174.600000  165.600000  ...   

         2010        2011        2012        2013        2014        2015  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1   86.208587   82.105187   78.354228   74.991597   71.996841   69.288947   
2   87.600000   83.900000   80.300000   76.800000   73.600000   70.400000   
3  121.783272  118.442691  115.498791  112.667923  110.264426  107.765705   
4  120.300000  112.300000  105.000000   98.600000   93.000000   88.200000   

         2016        2017       2018       2019  
0         NaN         NaN        NaN        NaN  
1   66.667614   64.347025  62.115387  60.098659  
2   67.600000   64.900000  62.500000  60.300000  
3  105.055552  102.430680  99.598781  96.814240  
4   84.200000   80.600000  77.700000  74.700000  

[5 rows x 24 columns]

In [477]:
mortality.isnull().sum()*100/mortality.shape[0]

Country Name      0.000000
Country Code      0.000000
Indicator Name    0.000000
Indicator Code    0.000000
2000              9.398496
2001              9.398496
2002              9.398496
2003              9.398496
2004              9.398496
2005              9.398496
2006              9.398496
2007              9.398496
2008              9.398496
2009              9.398496
2010              9.398496
2011              9.398496
2012              9.398496
2013              9.398496
2014              9.398496
2015              9.398496
2016              9.398496
2017              9.398496
2018              9.398496
2019              9.398496
dtype: float64

#### Since there are 9.40% nulls, these can also be dropped in the table at it has neglible effect in the aggregated and summarized analysis. 

In [478]:
cleaned_mortality = mortality.dropna(axis=0)
cleaned_mortality.head()

Country Name Country Code  \
1  Africa Eastern and Southern          AFE   
2                  Afghanistan          AFG   
3   Africa Western and Central          AFW   
4                       Angola          AGO   
5                      Albania          ALB   

                                    Indicator Name Indicator Code        2000  \
1  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  136.917815   
2  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  128.700000   
3  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  169.548432   
4  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT  203.900000   
5  Mortality rate, under-5 (per 1,000 live births)    SH.DYN.MORT   27.200000   

         2001        2002        2003        2004        2005  ...  \
1  131.670422  126.139085  120.635993  115.263632  110.060055  ...   
2  124.600000  120.400000  116.300000  112.100000  107.900000  ...   
3  164.533824  159.251950  153.916285  148.577352  143.417384  ...   
4  197.800000  190.900000  183.200000  174.600000  165.600000  ...   
5   25.800000   24.400000   22.900000   21.400000   20.000000  ...   

         2010        2011        2012        2013        2014        2015  \
1   86.208587   82.105187   78.354228   74.991597   71.996841   69.288947   
2   87.600000   83.900000   80.300000   76.800000   73.600000   70.400000   
3  121.783272  118.442691  115.498791  112.667923  110.264426  107.765705   
4  120.300000  112.300000  105.000000   98.600000   93.000000   88.200000   
5   13.200000   12.100000   11.200000   10.400000    9.900000    9.600000   

         2016        2017       2018       2019  
1   66.667614   64.347025  62.115387  60.098659  
2   67.600000   64.900000  62.500000  60.300000  
3  105.055552  102.430680  99.598781  96.814240  
4   84.200000   80.600000  77.700000  74.700000  
5    9.400000    9.400000   9.500000   9.700000  

[5 rows x 24 columns]

In [479]:
cleaned_mortality.isnull().sum()*100/cleaned_mortality.shape[0]

Country Name      0.0
Country Code      0.0
Indicator Name    0.0
Indicator Code    0.0
2000              0.0
2001              0.0
2002              0.0
2003              0.0
2004              0.0
2005              0.0
2006              0.0
2007              0.0
2008              0.0
2009              0.0
2010              0.0
2011              0.0
2012              0.0
2013              0.0
2014              0.0
2015              0.0
2016              0.0
2017              0.0
2018              0.0
2019              0.0
dtype: float64

In [480]:
cleaned_mortality['Country Name'].unique()

array(['Africa Eastern and Southern', 'Afghanistan',
       'Africa Western and Central', 'Angola', 'Albania', 'Andorra',
       'Arab World', 'United Arab Emirates', 'Argentina', 'Armenia',
       'Antigua and Barbuda', 'Australia', 'Austria', 'Azerbaijan',
       'Burundi', 'Belgium', 'Benin', 'Burkina Faso', 'Bangladesh',
       'Bulgaria', 'Bahrain', 'Bahamas, The', 'Bosnia and Herzegovina',
       'Belarus', 'Belize', 'Bolivia', 'Brazil', 'Barbados',
       'Brunei Darussalam', 'Bhutan', 'Botswana',
       'Central African Republic', 'Canada',
       'Central Europe and the Baltics', 'Switzerland', 'Chile', 'China',
       "Cote d'Ivoire", 'Cameroon', 'Congo, Dem. Rep.', 'Congo, Rep.',
       'Colombia', 'Comoros', 'Cabo Verde', 'Costa Rica',
       'Caribbean small states', 'Cuba', 'Cyprus', 'Czech Republic',
       'Germany', 'Djibouti', 'Dominica', 'Denmark', 'Dominican Republic',
       'Algeria', 'East Asia & Pacific (excluding high income)',
       'Early-demographic dividen

In [481]:
cleaned_mortality.to_csv('Databases/cleaned_mortality.csv')

<hr>

### Causes of death in children under 5

In [482]:
cause = pd.read_csv('Databases/causes-of-death-in-children-under-5.csv')
cause.head(2)

Entity Code  Year  Invasive Non-typhoidal Salmonella (iNTS)    \
0  Afghanistan  AFG  1990                                   48.186866   
1  Afghanistan  AFG  1991                                   54.688521   

   Interpersonal violence    Nutritional deficiencies    Acute hepatitis    \
0                     105.0                      1779.0              718.0   
1                     130.0                      1822.0              741.0   

   Neoplasms    Measles    Digestive diseases    ...  \
0        431.0     8649.0                 477.0  ...   
1        439.0     8669.0                 495.0  ...   

   Other neonatal disorders    Whooping cough    Diarrheal diseases    \
0                      7112.0            2455.0                3968.0   
1                      7574.0            2385.0                4650.0   

   Fire, heat, and hot substances    Road injuries    Tuberculosis    \
0                             131.0            802.0           808.0   
1                             129.0            781.0           800.0   

   HIV/AIDS    Drowning    Malaria    Syphilis    
0        10.0       776.0       21.0  123.089256  
1        12.0       748.0       41.0  132.205263  

[2 rows x 32 columns]

In [483]:
cause.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8400 entries, 0 to 8399
Data columns (total 32 columns):
 #   Column                                                      Non-Null Count  Dtype  
---  ------                                                      --------------  -----  
 0   Entity                                                      8400 non-null   object 
 1   Code                                                        6150 non-null   object 
 2   Year                                                        8400 non-null   int64  
 3   Invasive Non-typhoidal Salmonella (iNTS)                    8220 non-null   float64
 4   Interpersonal violence                                      8220 non-null   float64
 5   Nutritional deficiencies                                    8220 non-null   float64
 6   Acute hepatitis                                             8220 non-null   float64
 7   Neoplasms                                                   8220 non-null   float64
 8 

In [484]:
cause.describe()

Year  Invasive Non-typhoidal Salmonella (iNTS)    \
count  8400.000000                                 8220.000000   
mean   2004.500000                                 2615.949104   
std       8.655957                                 9855.561015   
min    1990.000000                                    0.000054   
25%    1997.000000                                    0.050169   
50%    2004.500000                                    2.513477   
75%    2012.000000                                   78.027837   
max    2019.000000                                62334.445150   

       Interpersonal violence    Nutritional deficiencies    \
count               8220.000000                 8220.000000   
mean                 859.594161                14379.168127   
std                 2577.264853                51287.896913   
min                    0.000000                    0.000000   
25%                    2.000000                    1.000000   
50%                   19.000000                   57.000000   
75%                  231.250000                 2098.750000   
max                21223.000000               524103.000000   

       Acute hepatitis     Neoplasms        Measles    Digestive diseases    \
count        8220.000000   8220.000000    8220.000000           8220.000000   
mean         1749.517883   3173.361679   19596.436861           2709.428954   
std          6597.215893   9702.950682   74162.090312           8412.127212   
min             0.000000      0.000000       0.000000              0.000000   
25%             0.000000     11.000000       0.000000              4.000000   
50%             4.000000     98.000000       4.000000             60.000000   
75%            86.000000    776.000000    1554.750000            567.250000   
max         50184.000000  85197.000000  704288.000000          77952.000000   

       Cirrhosis and other chronic liver diseases    Chronic kidney disease    \
count                                   8220.000000                8220.00000   
mean                                     579.309489                 673.26472   
std                                     1847.469859                2073.84993   
min                                        0.000000                   0.00000   
25%                                        1.000000                   1.00000   
50%                                        9.000000                  16.00000   
75%                                       98.250000                 169.00000   
max                                    15916.000000               18047.00000   

       ...  Other neonatal disorders    Whooping cough    \
count  ...                 8220.000000       8220.000000   
mean   ...                25498.213017       8699.608273   
std    ...                83323.081241      28048.043215   
min    ...                    0.000000          0.000000   
25%    ...                   34.000000          0.000000   
50%    ...                  395.500000         49.000000   
75%    ...                 4382.250000       1291.250000   
max    ...               539952.000000     240021.000000   

       Diarrheal diseases    Fire, heat, and hot substances    \
count          8.220000e+03                       8220.000000   
mean           5.687576e+04                       1185.287348   
std            1.910459e+05                       3694.451853   
min            0.000000e+00                          0.000000   
25%            5.000000e+00                          3.000000   
50%            2.710000e+02                         36.500000   
75%            8.405250e+03                        307.000000   
max            1.649581e+06                      35583.000000   

       Road injuries    Tuberculosis       HIV/AIDS       Drowning    \
count      8220.000000     8220.000000    8220.000000    8220.000000   
mean       3699.817762     6598.908273    8091.167032    4954.366545   
std       11737.107220    22413.420663   30454.082266   17598.9085

In [485]:
cause.describe(include=object)

Entity  Code
count          8400  6150
unique          280   205
top     Afghanistan   AFG
freq             30    30

In [486]:
cause.isnull().sum()*100/cause.shape[0]

Entity                                                         0.000000
Code                                                          26.785714
Year                                                           0.000000
Invasive Non-typhoidal Salmonella (iNTS)                       2.142857
Interpersonal violence                                         2.142857
Nutritional deficiencies                                       2.142857
Acute hepatitis                                                2.142857
Neoplasms                                                      2.142857
Measles                                                        2.142857
Digestive diseases                                             2.142857
Cirrhosis and other chronic liver diseases                     2.142857
Chronic kidney disease                                         2.142857
Cardiovascular diseases                                        2.142857
Congenital birth defects                                       2

In [487]:
cause.duplicated().sum()

0

#### Since there are 2.14% nulls per cause of deaths, these can also be dropped in the table at it has neglible effect in the aggregated and summarized analysis. 

In [488]:
cause.dropna(axis=0, inplace=True)
cause.head()

Entity Code  Year  Invasive Non-typhoidal Salmonella (iNTS)    \
0  Afghanistan  AFG  1990                                   48.186866   
1  Afghanistan  AFG  1991                                   54.688521   
2  Afghanistan  AFG  1992                                   67.520172   
3  Afghanistan  AFG  1993                                   78.250360   
4  Afghanistan  AFG  1994                                   82.658468   

   Interpersonal violence    Nutritional deficiencies    Acute hepatitis    \
0                     105.0                      1779.0              718.0   
1                     130.0                      1822.0              741.0   
2                     155.0                      2069.0              836.0   
3                     178.0                      2427.0              970.0   
4                     194.0                      2649.0             1063.0   

   Neoplasms    Measles    Digestive diseases    ...  \
0        431.0     8649.0                 477.0  ...   
1        439.0     8669.0                 495.0  ...   
2        486.0     8539.0                 554.0  ...   
3        549.0     8949.0                 630.0  ...   
4        589.0    10642.0                 681.0  ...   

   Other neonatal disorders    Whooping cough    Diarrheal diseases    \
0                      7112.0            2455.0                3968.0   
1                      7574.0            2385.0                4650.0   
2                      8614.0            2370.0                5833.0   
3                      9458.0            2659.0                7800.0   
4                      9823.0            3187.0                7894.0   

   Fire, heat, and hot substances    Road injuries    Tuberculosis    \
0                             131.0            802.0           808.0   
1                             129.0            781.0           800.0   
2                             137.0            821.0           863.0   
3                             155.0            923.0           979.0   
4                             170.0           1015.0          1064.0   

   HIV/AIDS    Drowning    Malaria    Syphilis    
0        10.0       776.0       21.0  123.089256  
1        12.0       748.0       41.0  132.205263  
2        13.0       777.0       51.0  180.232202  
3        16.0       872.0       24.0  239.050138  
4        19.0       961.0       52.0  258.975281  

[5 rows x 32 columns]

#### Since 2020 is not available and the years below 2000 will not provide any sufficient recent conclusions and recommendations, these years will not be included or will be dropped to accomodate other databases which has lesser years as well. 

In [489]:
cause.drop(cause.index[cause['Year'] < 2000], inplace=True)
cleaned_cause = cause
cleaned_cause.head()

Entity Code  Year  Invasive Non-typhoidal Salmonella (iNTS)    \
10  Afghanistan  AFG  2000                                  130.895780   
11  Afghanistan  AFG  2001                                  133.656719   
12  Afghanistan  AFG  2002                                  139.812677   
13  Afghanistan  AFG  2003                                  181.026615   
14  Afghanistan  AFG  2004                                  203.233435   

    Interpersonal violence    Nutritional deficiencies    Acute hepatitis    \
10                     180.0                      2526.0             1104.0   
11                     179.0                      2426.0             1065.0   
12                     191.0                      2268.0             1011.0   
13                     219.0                      2478.0             1128.0   
14                     222.0                      2467.0             1129.0   

    Neoplasms    Measles    Digestive diseases    ...  \
10        595.0    13818.0                 716.0  ...   
11        585.0    13944.0                 715.0  ...   
12        571.0    13266.0                 705.0  ...   
13        680.0     4295.0                 821.0  ...   
14        714.0     1735.0                 854.0  ...   

    Other neonatal disorders    Whooping cough    Diarrheal diseases    \
10                     10269.0            4154.0               10960.0   
11                     10557.0            3962.0               11073.0   
12                     11001.0            3820.0               11045.0   
13                     11526.0            4241.0               12009.0   
14                     11800.0            4095.0               11925.0   

    Fire, heat, and hot substances    Road injuries    Tuberculosis    \
10                             180.0           1049.0          1048.0   
11                             173.0            994.0          1000.0   
12                             161.0            915.0           919.0   
13                             189.0           1080.0          1006.0   
14                             195.0           1107.0           992.0   

    HIV/AIDS    Drowning    Malaria    Syphilis    
10        30.0      1036.0       71.0  267.957457  
11        30.0       988.0       54.0  258.439617  
12        31.0       906.0      740.0  267.007930  
13        32.0      1078.0      589.0  283.043790  
14        33.0      1117.0      209.0  280.265432  

[5 rows x 32 columns]

In [490]:
cleaned_cause = cleaned_cause.rename(columns={"Entity": "Country Name", "Code": "Country Code"})
cleaned_cause.head()

Country Name Country Code  Year  \
10  Afghanistan          AFG  2000   
11  Afghanistan          AFG  2001   
12  Afghanistan          AFG  2002   
13  Afghanistan          AFG  2003   
14  Afghanistan          AFG  2004   

    Invasive Non-typhoidal Salmonella (iNTS)    Interpersonal violence    \
10                                  130.895780                     180.0   
11                                  133.656719                     179.0   
12                                  139.812677                     191.0   
13                                  181.026615                     219.0   
14                                  203.233435                     222.0   

    Nutritional deficiencies    Acute hepatitis    Neoplasms    Measles    \
10                      2526.0             1104.0        595.0    13818.0   
11                      2426.0             1065.0        585.0    13944.0   
12                      2268.0             1011.0        571.0    13266.0   
13                      2478.0             1128.0        680.0     4295.0   
14                      2467.0             1129.0        714.0     1735.0   

    Digestive diseases    ...  Other neonatal disorders    Whooping cough    \
10                 716.0  ...                     10269.0            4154.0   
11                 715.0  ...                     10557.0            3962.0   
12                 705.0  ...                     11001.0            3820.0   
13                 821.0  ...                     11526.0            4241.0   
14                 854.0  ...                     11800.0            4095.0   

    Diarrheal diseases    Fire, heat, and hot substances    Road injuries    \
10               10960.0                             180.0           1049.0   
11               11073.0                             173.0            994.0   
12               11045.0                             161.0            915.0   
13               12009.0                             189.0           1080.0   
14               11925.0                             195.0           1107.0   

    Tuberculosis    HIV/AIDS    Drowning    Malaria    Syphilis    
10          1048.0        30.0      1036.0       71.0  267.957457  
11          1000.0        30.0       988.0       54.0  258.439617  
12           919.0        31.0       906.0      740.0  267.007930  
13          1006.0        32.0      1078.0      589.0  283.043790  
14           992.0        33.0      1117.0      209.0  280.265432  

[5 rows x 32 columns]

In [491]:
cleaned_cause.isnull().sum()*100/cleaned_cause.shape[0]

Country Name                                                  0.0
Country Code                                                  0.0
Year                                                          0.0
Invasive Non-typhoidal Salmonella (iNTS)                      0.0
Interpersonal violence                                        0.0
Nutritional deficiencies                                      0.0
Acute hepatitis                                               0.0
Neoplasms                                                     0.0
Measles                                                       0.0
Digestive diseases                                            0.0
Cirrhosis and other chronic liver diseases                    0.0
Chronic kidney disease                                        0.0
Cardiovascular diseases                                       0.0
Congenital birth defects                                      0.0
Lower respiratory infections                                  0.0
Neonatal p

In [492]:
cleaned_cause['Country Name'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'American Samoa', 'Andorra',
       'Angola', 'Antigua and Barbuda', 'Argentina', 'Armenia',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain',
       'Bangladesh', 'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin',
       'Bermuda', 'Bhutan', 'Bolivia', 'Bosnia and Herzegovina',
       'Botswana', 'Brazil', 'Brunei', 'Bulgaria', 'Burkina Faso',
       'Burundi', 'Cambodia', 'Cameroon', 'Canada', 'Cape Verde',
       'Central African Republic', 'Chad', 'Chile', 'China', 'Colombia',
       'Comoros', 'Congo', 'Cook Islands', 'Costa Rica', "Cote d'Ivoire",
       'Croatia', 'Cuba', 'Cyprus', 'Czechia',
       'Democratic Republic of Congo', 'Denmark', 'Djibouti', 'Dominica',
       'Dominican Republic', 'Ecuador', 'Egypt', 'El Salvador',
       'Equatorial Guinea', 'Eritrea', 'Estonia', 'Eswatini', 'Ethiopia',
       'Fiji', 'Finland', 'France', 'Gabon', 'Gambia', 'Georgia',
       'Germany', 'Ghana', 'Greece', 'Greenland', 'G

In [493]:
cleaned_cause.to_csv('Databases/cleaned_cause.csv')

<hr>

### Malnutrition across the globe (Data of countries from 1983-2019)

In [494]:
malnutrition = pd.read_csv('Databases/malnutrition_country_avg.csv')
malnutrition.head()

Country  Income Classification  Severe Wasting    Wasting  Overweight  \
0  AFGHANISTAN                      0        3.033333  10.350000    5.125000   
1      ALBANIA                      2        4.075000   7.760000   20.800000   
2      ALGERIA                      2        2.733333   5.942857   12.833333   
3       ANGOLA                      1        2.400000   6.933333    2.550000   
4    ARGENTINA                      2        0.200000   2.150000   11.125000   

    Stunting  Underweight  U5 Population ('000s)  
0  47.775000    30.375000            4918.561500  
1  24.160000     7.700000             232.859800  
2  19.571429     7.342857            3565.213143  
3  42.633333    23.600000            3980.054000  
4  10.025000     2.600000            3613.651750

In [495]:
malnutrition.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 152 entries, 0 to 151
Data columns (total 8 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Country                152 non-null    object 
 1   Income Classification  152 non-null    int64  
 2   Severe Wasting         140 non-null    float64
 3   Wasting                150 non-null    float64
 4   Overweight             149 non-null    float64
 5   Stunting               151 non-null    float64
 6   Underweight            150 non-null    float64
 7   U5 Population ('000s)  152 non-null    float64
dtypes: float64(6), int64(1), object(1)
memory usage: 9.6+ KB


In [496]:
malnutrition.describe()

Income Classification  Severe Wasting     Wasting  Overweight  \
count             152.000000      140.000000  150.000000  149.000000   
mean                1.427632        2.168650    6.599257    7.201638   
std                 0.967019        1.708939    4.481723    4.649144   
min                 0.000000        0.000000    0.000000    0.962500   
25%                 1.000000        0.900000    3.262500    3.850000   
50%                 1.000000        1.872500    5.710714    6.300000   
75%                 2.000000        2.822727    8.740476    9.080000   
max                 3.000000       11.400000   23.650000   26.500000   

         Stunting  Underweight  U5 Population ('000s)  
count  151.000000   150.000000             152.000000  
mean    25.814728    13.503047            4042.927052  
std     14.686807    10.895839           13164.191928  
min      1.000000     0.100000               1.000000  
25%     13.485000     4.305000             241.765813  
50%     24.160000    10.380000             981.233486  
75%     36.564935    19.496875            3002.433080  
max     57.600000    46.266667          123014.491000

In [497]:
malnutrition.describe(include=object)

Country
count           152
unique          152
top     AFGHANISTAN
freq              1

In [498]:
malnutrition.isnull().sum()*100/malnutrition.shape[0]

Country                  0.000000
Income Classification    0.000000
Severe Wasting           7.894737
Wasting                  1.315789
Overweight               1.973684
Stunting                 0.657895
Underweight              1.315789
U5 Population ('000s)    0.000000
dtype: float64

In [499]:
malnutrition.duplicated().sum()

0

In [500]:
malnutrition.dropna(axis=0, inplace=True)
cleaned_malnutrition = malnutrition
cleaned_malnutrition = cleaned_malnutrition.rename(columns={"Country": "Country Name"})
cleaned_malnutrition['Country Name'] = cleaned_malnutrition['Country Name'].str.title()
cleaned_malnutrition.head()

Country Name  Income Classification  Severe Wasting    Wasting  Overweight  \
0  Afghanistan                      0        3.033333  10.350000    5.125000   
1      Albania                      2        4.075000   7.760000   20.800000   
2      Algeria                      2        2.733333   5.942857   12.833333   
3       Angola                      1        2.400000   6.933333    2.550000   
4    Argentina                      2        0.200000   2.150000   11.125000   

    Stunting  Underweight  U5 Population ('000s)  
0  47.775000    30.375000            4918.561500  
1  24.160000     7.700000             232.859800  
2  19.571429     7.342857            3565.213143  
3  42.633333    23.600000            3980.054000  
4  10.025000     2.600000            3613.651750

In [501]:
cleaned_malnutrition.isnull().sum()*100/cleaned_malnutrition.shape[0]

Country Name             0.0
Income Classification    0.0
Severe Wasting           0.0
Wasting                  0.0
Overweight               0.0
Stunting                 0.0
Underweight              0.0
U5 Population ('000s)    0.0
dtype: float64

In [502]:
cleaned_malnutrition['Country Name'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Angola', 'Argentina',
       'Armenia', 'Australia', 'Azerbaijan', 'Bangladesh', 'Barbados',
       'Belarus', 'Belize', 'Benin', 'Bhutan', 'Bolivia',
       'Bosnia And Herzegovina', 'Botswana', 'Brazil',
       'Brunei Darussalam', 'Bulgaria', 'Burkina Faso', 'Burundi',
       'Cambodia', 'Cameroon', 'Central African Republic', 'Chad',
       'China', 'Colombia', 'Comoros', 'Congo', "Cote D'Ivoire",
       'Czechia', "Democratic People'S Rep. Of Korea",
       'Democratic Rep. Of The Congo', 'Djibouti', 'Dominican Republic',
       'Ecuador', 'Egypt', 'El Salvador', 'Equatorial Guinea', 'Eritrea',
       'Eswatini', 'Ethiopia', 'Fiji', 'Gabon', 'Gambia', 'Georgia',
       'Germany', 'Ghana', 'Guatemala', 'Guinea', 'Guinea-Bissau',
       'Guyana', 'Haiti', 'Honduras', 'India', 'Indonesia', 'Iran',
       'Iraq', 'Jamaica', 'Japan', 'Jordan', 'Kazakhstan', 'Kenya',
       'Kuwait', 'Kyrgyzstan', "Lao People'S Democratic Rep.", 'Lebanon',
  

In [503]:
countries = pd.read_csv('Databases/countrynamesncodes.csv')
countries.head()

Country Name Country Code
0     Afghanistan          AFG
1         Albania          ALB
2         Algeria          DZA
3  American Samoa          ASM
4         Andorra          AND

#### Since the table does not contain a country code column, the best solution was to merge with a country code dataset from IBAN using Alpha-3 coding. `See datasets section for the source reference.`

In [504]:
cleaned_malnutrition = pd.merge(cleaned_malnutrition, countries, how='left', on='Country Name')
column_to_reorder = cleaned_malnutrition.pop('Country Code')
cleaned_malnutrition.insert(1, 'Country Code', column_to_reorder)
cleaned_malnutrition.head()

Country Name Country Code  Income Classification  Severe Wasting    Wasting  \
0  Afghanistan          AFG                      0        3.033333  10.350000   
1      Albania          ALB                      2        4.075000   7.760000   
2      Algeria          DZA                      2        2.733333   5.942857   
3       Angola          AGO                      1        2.400000   6.933333   
4    Argentina          ARG                      2        0.200000   2.150000   

   Overweight   Stunting  Underweight  U5 Population ('000s)  
0    5.125000  47.775000    30.375000            4918.561500  
1   20.800000  24.160000     7.700000             232.859800  
2   12.833333  19.571429     7.342857            3565.213143  
3    2.550000  42.633333    23.600000            3980.054000  
4   11.125000  10.025000     2.600000            3613.651750

In [505]:
cleaned_malnutrition.isnull().sum()*100/cleaned_malnutrition.shape[0]

Country Name              0.000000
Country Code             17.142857
Income Classification     0.000000
Severe Wasting            0.000000
Wasting                   0.000000
Overweight                0.000000
Stunting                  0.000000
Underweight               0.000000
U5 Population ('000s)     0.000000
dtype: float64

#### Some countries will not be included in the possible merges because they are not given country codes, but these will not be dropped as the malnutrition table will mostly be utilized for determining the income classification of each country. 

In [506]:
cleaned_malnutrition.to_csv('Databases/cleaned_malnutrition.csv')

<hr>
<hr>

## Data Visualization